### PR TITLE
return Into value in HasIntoDefined() in SelectStatement

### PR DESCRIPTION
**Pull request by jennifersp**
This allows to get Into value without doing switch on its types.

### CODE REVIEW

File: go/vt/sqlparser/ast.go [hunk 1]
```diff
@@ -421,7 +421,7 @@ type SelectStatement interface {
 	SetLock(string)
 	SetOrderBy(OrderBy)
 	SetInto(*Into) error
-	HasIntoDefined() bool
+	HasIntoDefined() (*Into, bool)
```
Comment (fulghum) [inline go/vt/sqlparser/ast.go:424]: Feels a little non-standard to have a `Has` method that returns a field and a boolean value – from that prefix, I'd expect it to return a boolean value. How about renaming the method to `GetInto` and just having it return `*Into`? That would be a more common pattern and it would still be easy to test if the result == nil in the places where you were using the boolean.

Comment (jennifersp) [reply go/vt/sqlparser/ast.go]: You're right! It makes sense to have GetInto() and use the result to check whether it's defined.

File: go/vt/sqlparser/sql.y [hunk 2]
```diff
@@ -699,7 +699,7 @@ common_table_expression:
 union_lhs:
   base_select
   {
-    if _, ok := $1.HasIntoDefined(); ok {
+    if $1.GetInto() == nil {
```
Comment (fulghum) [inline go/vt/sqlparser/sql.y]: Should these next three instances be `!=` instead of `==`? Good chance to check if we have tests that fired, and if not, see if there are test gaps we should fix.

Comment (jennifersp) [reply go/vt/sqlparser/sql.y]: yea, I made typos and it was caught in the previous commit. I fixed it in the latest commit.

**Review by fulghum — APPROVED** [review]
👍🏼
